### ISSUE TITLE
32bit release

### ISSUE THREAD

**Issue by MyTDT-Mysoft** [ange-yaghi/engine-sim#110]
why the release is 64bit? when this could run confortable in 32bit... just to deliberatebly sabotage compatibility? or it was plain laziness?

**Comment by Najsr** [comment]
It is apparently planned: https://github.com/ange-yaghi/engine-sim/issues/95 But IMHO 32bit is obsolete nowadays at least on x86 CPUs. Who uses it on modern day PCs.

**Comment by NexusXe** [comment]
Yeah, I couldn't really see a purpose for a 32-bit release. Maybe for emulation with Wine? But as seen [here](https://forum.winehq.org/viewtopic.php?t=15568), there isn't even a performance improvement.

**Comment by MyTDT-Mysoft** [comment]
> It is apparently planned: #95 But IMHO 32bit is obsolete nowadays at least on x86 CPUs. Who uses it on modern day PCs.

first... check your definition of obsolete... second... ME.... even that i have a 64bit CPU... me OS is and will always be 32bit...

**Comment by MyTDT-Mysoft** [comment]
> Yeah, I couldn't really see a purpose for a 32-bit release. Maybe for emulation with Wine? But as seen [here](https://forum.winehq.org/viewtopic.php?t=15568), there isn't even a performance improvement.

well its for windows... windows (even 64bit) runs 32bit programs just fine (its not linux)... ALL window's basically can... so forcing to use 64bit for something that does not need it (and almost nothing need it...) is just not nice :)

**Comment by NexusXe** [comment]
> well its for windows... windows (even 64bit) runs 32bit programs just fine (its not linux)... ALL window's basically can... so forcing to use 64bit for something that does not need it (and almost nothing need it...) is just not nice :)

right... but this program depends on things that only 64-bit capable computers will reasonably have, such as a powerful enough processor or Vulkan support. It's as simple as that, really. You'd be hard pressed to find someone who needs such support, with the downside that adding it makes developing quite a bit more complex.

Do you yourself have a requirement for this program to be 32-bit? If so, please elaborate on why.

**Comment by MyTDT-Mysoft** [comment]
> > well its for windows... windows (even 64bit) runs 32bit programs just fine (its not linux)... ALL window's basically can... so forcing to use 64bit for something that does not need it (and almost nothing need it...) is just not nice :)
> 
> right... but this program depends on things that only 64-bit capable computers will reasonably have, such as a powerful enough processor. It's as simple as that, really. You'd be hard pressed to find someone who needs such support, with the downside that adding it makes developing quite a bit more complex.
> 
> Do you yourself have a requirement for this program to be 32-bit? If so, please elaborate on why.

well the processor is 64bit capable... but the thing is newer computer can also have 32bit OSes even being 64bit+32bit processors and you're not getting any speed up anyway by using 64bit here (not anything significant at least), contrary to what would be possible if this program was rewritten a programmer (no offense but while i value the nice physics involved, the code is way slower than it could be (thats the ONLY reason it actually require a powerful enough processor to begin with)...
but no i just wanted to try it here... but i can't since theres no 32bit releases... i guess it would be reasonable for me to just compile it for 32bit if thats doable with mingw... but otherwise i can't even try this, and i wont downgrade my OS just for it hehe

my point is just that since windows is 32bit capable and windows usually aims for compatibility and this dont have any marginal good reason to not have a 32bit binary for windows that support 32bits binaries everywhere... i'm calling it a bug (just advocating for those who get discriminated by not wanting to go with latest almost useless trends)

**Comment by NexusXe** [comment]
64 bit programs are faster and more capable. Windows x86_64 doesn't "support" 32-bit applications. It emulates them, using a translator called "WOW64."

This translator causes overhead and slowdown. Not a lot, but some.

I almost promise you that you're not using a 32-bit operating system, assuming that you have more than 4 gigabytes of RAM.

64 bit binaries, for all intents and purposes, are faster.

Regardless of speed, users who need this 32-bit support are a non-issue, because there are virtually none.

**Comment by MyTDT-Mysoft** [comment]
> 64 bit programs are faster and more capable. Windows x86_64 doesn't "support" 32-bit applications. It emulates them, using a translator called "WOW64."
> 
> This translator causes overhead and slowdown. Not a lot, but some.
> 
> I almost promise you that you're not using a 32-bit operating system, assuming that you have more than 4 gigabytes of RAM.
> 
> 64 bit binaries, for all intents and purposes, are faster.
> 
> Regardless of speed, users who need this 32-bit support are a non-issue, because there are virtually none.

![image](https://user-images.githubusercontent.com/16870416/184911311-ef106edd-50d4-4cda-a210-04ccbb22329a.png)

ahem... yes its windows 7 32bit... with 8gb of ram.... and i can use all of the 8gb... in fact the only reason i have 8gb basically is for chrome that ends using too much lol...

again... "WINE" says "wine is not an emulator"... and you think WOW64 is?? its a 32bit layer (not even a translator anyway, and it have less overheard than the C++ "safety features") on top of 64bit... its running on the native processor... calling the native 64bit kernel... and etc... etc...

and no 64bit programs are not really faster or more capable... usually the difference is so insignificant that 32bit programs compiled by GCC (a faster better compiler than MSVC) tend to outperform 64bit binaries, theres some cases that are most optimal for 64bit CPU (like using 64bit integers without SSE/AVX, but this software as unoptimized as it is that is FAAAR from being even marginally a difference)

**Comment by MyTDT-Mysoft** [comment]
> You use an end-of-life operating system. Expecting compatibility is a bit far-fetched. If you require this program you can always compile it yourself, but expecting an open source program made in someone's free time to be compatible with your end-of-life, outdated, and obsolete operating system is unrealistic.

well i expect common sense out of programmers... just that... specially since even if this was windows 10 (why i would delireberately slowdown my whole machine to use that junk?) it would still be the better 32bit and it would still have the same problem... so making excuses does not help :P

**Comment by MyTDT-Mysoft** [comment]
> (also, ps: advertising that you use a cracked operating system is typically not advisable.. lol)
lol first... it was unsupported..... and now you care? make up your mind :P

**Comment by otDan** [comment]
go ahead and do a pr to the project if you really are in need of a 32bit for your obsolete system then, if you don't want to run "junk" on your pc then support everything that you need yourself, think that's the best solution for you with that mentality...

**Comment by MyTDT-Mysoft** [comment]
> go ahead and do a pr to the project if you really are in need of a 32bit for your obsolete system then, if you don't want to run "junk" on your pc then support everything that you need yourself, think that's the best solution for you with that mentality...

again.... check the definition of obsolete lol... the system is totally not obsolete... the mindset of thinking that is what is wrong... and the thing here... my job is just report bugs when a program does not run on windows when they should
and so i did... i may check if the compile will work later, because i guess it should work... despite that i would need to reinstall that "joke" of program that cmake is. Or maybe i should just create a makefile or a batch myself since cmake joke can't do that.

**Comment by IonicOwl** [comment]
Seems to me like you have an axe to grind, and here really isn't the place to do it.
The software is unfinished and not running as efficiently as it could be.  The author explicitly states this in the readme, I suggest you readit.
If you think you can bring improvements to the project then contribute, otherwise, please take your biases somewhere else.

**Comment by otDan** [comment]
> > go ahead and do a pr to the project if you really are in need of a 32bit for your obsolete system then, if you don't want to run "junk" on your pc then support everything that you need yourself, think that's the best solution for you with that mentality...
> 
> again.... check the definition of obsolete lol... the system is totally not obsolete... the mindset of thinking that is what is wrong... and the thing here... my job is just report bugs when a program does not run on windows when they should and so i did... i may check if the compile will work later, because i guess it should work... despite that i would need to reinstall that "joke" of program that cmake is. Or maybe i should just create a makefile or a batch myself since cmake joke can't do that.

I do not know what meaning you attribute to obsolete, but for me it's something that is not cutting edge anymore and is not supported, from what I can see on your github most of your issues are always relating not having available 32bit versions for stuff and yet you say it's still perfectly usable as a system for these years.

**Comment by MyTDT-Mysoft** [comment]
> 

yes because other people are not taking the battle serious... and using garbage without reason... yet the system is perfectly usable i have everything that i need here... but it does not mean i wont try to make people see the reason when they contribute to something that is bad... as this isnt APPLE... and since windows 10 still support 32bit... 32bit should still be supported... and so on... so all projects that do stuff by laziness i open bug reports to be fixed
because i'm tired of fixing everything for everyone that is too lazy to plan ahead for good sensible tartgets

and calling my windows obsolete is as offensive as me calling your mom obsolete... because they are not cutting edge which is irrelevant... since they are fully capable :)

**Comment by Najsr** [comment]
> > 
> 
> yes because other people are not taking the battle serious... and using garbage without reason... yet the system is perfectly usable i have everything that i need here... but it does not mean i wont try to make people see the reason when they contribute to something that is bad... as this isnt APPLE... and since windows 10 still support 32bit... 32bit should still be supported... and so on... so all projects that do stuff by laziness i open bug reports to be fixed because i'm tired of fixing everything for everyone that is too lazy to plan ahead for good sensible tartgets
> 
> and calling my windows obsolete is as offensive as me calling your mom obsolete... because they are not cutting edge which is irrelevant... since they are fully capable :)

Bringing mother into tech discussion. You are lacking arguments I guess.

By seeing your DxDiag screenshot you are using a CPU that is over decade old on a OS that is marked as EOL and no longer receives any security upgrades. So yes, both your PC and OS are obsolete.
Windows 10 does indeed support 32 bit version, but who on Earth has a enough capable system to run Win10 but uses 32 bit version. Even many GNU/Linux distributions don't support 32 bit since nearly nobody uses those.

**Comment by IonicOwl** [comment]
> 

What battle are you talking about?  Microsoft may choose to support 32-bit applications, that's their prerogative, doesn't mean someone writing software and being gracious enough to share its source code has to.  You say you have everything you need in the 32-bit world, and yet here you are.

**Comment by MyTDT-Mysoft** [comment]
> > > 
> > 
> > 
> > yes because other people are not taking the battle serious... and using garbage without reason... yet the system is perfectly usable i have everything that i need here... but it does not mean i wont try to make people see the reason when they contribute to something that is bad... as this isnt APPLE... and since windows 10 still support 32bit... 32bit should still be supported... and so on... so all projects that do stuff by laziness i open bug reports to be fixed because i'm tired of fixing everything for everyone that is too lazy to plan ahead for good sensible tartgets
> > and calling my windows obsolete is as offensive as me calling your mom obsolete... because they are not cutting edge which is irrelevant... since they are fully capable :)
> 
> Bringing mother into tech discussion. You are lacking arguments I guess.
> 
> By looking at your DxDiag screenshot you are using a CPU that is over decade old on a OS that is marked as EOL and no longer receives any security upgrades. So yes, both your PC and OS are obsolete. Windows 10 does indeed support 32 bit version, but who on Earth has a enough capable system to run Win10 but uses 32 bit version. Even many GNU/Linux distributions don't support 32 bit since nearly nobody uses those.
> 
> So why bother supporting 2 different bit versions when you can just support 64 bit and be just fine?

yeah well.. because "newer" doesnt mean much... it wouldnt change much having a newer CPU... but on the other hand i have a high end old gfx card... and SSD and everything... just the CPU that is low i have also a notebook that i use for testing stuff to make sure they work on win32, but that notebook despite being newer than this desktop it crawls because of win10 hehe, and yes... 
i wouldnt change or stop supporting something that is 10 years old yet... perfectly capable, but i respect my machine and it respect me... when it's gone i will probabily change to something 2019... that is probabily the newest year that i would be able to find compatible enough hardware... and after that... only ReactOS will have any chance at being windows... since even MS is trying to kil it due having a CEO that lives in the clouds...

but yeah as i said i have everything i need. but i would be interesting to try this... and since this should work fine on 32bit and since i basically dont even see any need to have it as 64bit... thats why i'm here, and if there would be only one then that one should be 32bit software (even for the 64bit OS) because they works as good and as capable on both places anything else is just people discrimination :)

against older stuff... which is totally not OBSOLETE, and it isnt like it requries any considerable amount of effort to "support" 32bit anyway, but i can't lift the weight of the whole ignorant world on my shoulders, so advocating...

**Comment by otDan** [comment]
> oh and most of time... when i do those on github, is because they say they have releases for windows... and yet they dont work on windows :)

"Windows 7 support ended on January 14, 2020" yes it does, on supported new versions :)

**Comment by MyTDT-Mysoft** [comment]
> > oh and most of time... when i do those on github, is because they say they have releases for windows... and yet they dont work on windows :)
> 
> "Windows 7 support ended on January 14, 2020" yes it does, on supported new versions :)

nope it does not work on windows 8... or windows 10 either ... it only works on windows 64bit..

**Comment by MyTDT-Mysoft** [comment]
> Imagine making this issue to blame the owner for being "lazy" and "deliberately sabotaging compatibility" because his **personal project** doesn't support 32 bit ... **even though** the owner has already created an issue with the intent of adding 32 bit support. #95. This level of entitlement is disgusting

no, thats how it usually goes (on gitbuts), but what you described stopped waaay sooner when it was pointed out that there was plans for that, after that was just "picking on me" for wanting that :) (and me tagging along)

**Comment by ffredyk** [comment]
@MyTDT-Mysoft 

Just out of curiosity, do you have any credible resources backing your words? Subjectively my system went from smooth to a lot more smoother when I transitioned from Win7 to Win10. I had perceived a lot more stability and performance raise from going to 32bit to 64bit. And when I upgraded my CPU to a latest most powerful architecture, my options grew exponentially (either by using 100 tabs in chrome, or having 20 applications running in the background, to most basic video rendering speed)

And don't take me wrong, I really didn't studied the performance comparisons between 32bit and 64bit and I am interested in this topic

**Comment by MyTDT-Mysoft** [comment]
> @MyTDT-Mysoft 
> 
> Just out of curiosity, do you have any credible resources backing your words? Subjectively my system went from smooth to a lot more smoother when I transitioned from Win7 to Win10. I had perceived a lot more stability and performance raise from going to 32bit to 64bit. And when I upgraded my CPU to a latest most powerful architecture, my options grew exponentially (either by using 100 tabs in chrome, or having 20 applications running in the background, to most basic video rendering speed)
> 
> And don't take me wrong, I really didn't studied the performance comparisons between 32bit and 64bit and I am interested in this topic

  Well i would need to check if theres a "single" resource, because its most bits here and there, and complaints from people that i've been helping, but theres other claiming the same as you.
Most of them claiming performance raise had a few things in common... they CHANGED from older PCs with tired HDDs to newer PCs that came with SSD, and usually went from 2gb of ram to 8gb or 16gb of ram, and almost nobody have 8gb on 32bit windows, because without the PAE patch you're limited to ~3.25gb usually (which is something that microsoft should had removed by itself (the need to PAE patch..) like it was on XP SP2, before they hindered due to drivers problems that does not exist anymore)
  Another factor is that indeed newer computers got faster RAM and faster speed, like for example i have a slow first generation I5, but newer generation I5 can be 2x or 3x faster (specially for programs that can use AVX and AVX512)
but usually you also had change from a beaten up OS, with a lot of garbage installed, with a polluted registry to a clean install, which always make things fast...
  Theres another factor that some things ACTUALLY improved in windows 10 kernel... (i wouldnt mind having windows 10 KERNEL here, without the shell, without metro, and without the new services and bloat, for example... but thats near impossible)
  Also if you want a comparsion try to run windows 7 with 1gb of ram vs win10 with 1gb... or compare win7 on a slow HDD vs win10 on a slow HDD... my VM with win10 took 30 seconds to open a control panel, because the vm "only had" 1gb of ram which is absurd, considering the QEMU (processor emulated), XP VM... running on a 2004 computer with only 768mb of ram... and that i allocated just 64mb of ram... was running way better than that (but ofcourse that XP i tunned to requrie just 16mb of ram to boot, by removing everything (even USB) because the VM would not need that)

and last but not least... chrome improved itself in between, and having a new (clean) chrome installed also helped for people, and in fact i have over 100 tabs open here on my 32bit win7, i actually did put 8gb just to be able to do that hehe, but... when i changed from XP to 7 i had to change from 4gb to 6gb just to keep up with the exact same load had on XP... and 64bit win7 would require 8gb just to keep up with that as well

one thing that they ACTUALLY improved on win7 and then again on win10, was "perceived" boot time, because XP used to do everything ahead of time, so that when it finishes its fully booted... win7 still have the HW booting ahead of time (including fully initializing the wifi that is slow), but win7 had services and tasks performed as low priority, low HD speed in background, and win10 initialized the HW in parallel (which would not easily work with older hw), and left many stuff to be loaded on demand (the first time its required, and even later in background), among other stuff, which improved the "restore from sleep" as well because of the async HW initializiation and so on...

another thing that maybe improved was windows defender... but i never have that one enabled... and probabily a bunch of stuff that i had disabled on win7, but that win10 make it extra harder to be able to disable... and overall other drivers improved... some of those you can get newer versions for win7 other improved naturally (better compiler and stuff) but only released to win10, which then again is not win7 fault... and they didnt improved because of anything that win10 did either...

but i will check if i can find extra sources with similar stuff, but most of those are hidden over the web behind a pile of ignorance from people that would ask when there will be a 128bit OS, and marketing and efforts to make 32bit to look bad, trying to force people to spend money on newer stuff.

but for me that does not really matter... because i only restart/power up my computer when theres a power outage

**Comment by ISAKEM** [comment]
Idk what to say, I have never seen such a dedication to try and prove that 32-bit is still relevant in 2022. I mean there is just no reason to run anything 32-bit in this day and age.
From like a retroperspective sure, I also like running older hardware/software 4fun. Im quite the OG Xbox fan, which is a 32-bit system. Running a 32-bit OS that is also EOL on your MAIN PC tho, idk bro u seem kinda crazy about this. 🤷‍♂️

**Comment by MyTDT-Mysoft** [comment]
> Idk what to say, I have never seen such a dedication to try and prove that 32-bit is still relevant in 2022. I mean there is just no reason to run anything 32-bit in this day and age. From like a retroperspective sure, I also like running older hardware/software 4fun. Im quite the OG Xbox fan, which is a 32-bit system. Running a 32-bit OS that is also EOL on your MAIN PC tho, idk bro u seem kinda crazy about this. man_shrugging

you can change to latest win10 with 32bit if you want... it wont change anything on the principle :P

**Comment by ISAKEM** [comment]
> > Idk what to say, I have never seen such a dedication to try and prove that 32-bit is still relevant in 2022. I mean there is just no reason to run anything 32-bit in this day and age. From like a retroperspective sure, I also like running older hardware/software 4fun. Im quite the OG Xbox fan, which is a 32-bit system. Running a 32-bit OS that is also EOL on your MAIN PC tho, idk bro u seem kinda crazy about this. man_shrugging
> 
> you can change to latest win10 with 32bit if you want... it wont change anything on the principle :P

Yeah bro sure I just wont be able to play literally any games anymore and I wont have more then 4 GB RAM, thats surely an enjoyable experience 🤣🤣🤣. Oh and lets not talk about the fact that this also includes GPU VRAM so any modern card cant even be used bcs the VRAM amount is bigger then 4GB XDDDDDD 🤥🤥🤥🤥

**Comment by MyTDT-Mysoft** [comment]
> > > Idk what to say, I have never seen such a dedication to try and prove that 32-bit is still relevant in 2022. I mean there is just no reason to run anything 32-bit in this day and age. From like a retroperspective sure, I also like running older hardware/software 4fun. Im quite the OG Xbox fan, which is a 32-bit system. Running a 32-bit OS that is also EOL on your MAIN PC tho, idk bro u seem kinda crazy about this. man_shrugging
> > 
> > 
> > you can change to latest win10 with 32bit if you want... it wont change anything on the principle :P
> 
> Yeah bro sure I just wont be able to play literally any games anymore and I wont have more then 4 GB RAM, thats surely an enjoyable experience roflroflrofl. Oh and lets not talk about the fact that this also includes GPU VRAM so any modern card cant even be used bcs the VRAM amount is bigger then 4GB XDDDDDD lying_facelying_facelying_facelying_face

you're back to that again... 32bit x86 is capable to go as up as 64gb of ram... and you dont handle the VRAM directly and even your 4gb card does not use require 4gb of address space either... for example i have 8gb of ram and 4gb of vram here... (on my 32bit OS) and i can use all the RAM (with up to 3gb per process at same time (process... not application)) and almost none of that address space is mapped to use VRAM even if i would use all the 4gb of it... 

and this isnt any magic... is just that ignorance is passed ahead and make people believe on crazy things, that are mere small conveniences

**Comment by Jckf** [comment]
This discussion has gotten out of hand. Closing.

**Comment by ISAKEM** [comment]
XD Clueless 32 bit man down

**Comment by MyTDT-Mysoft** [comment]
> XD Clueless 32 bit man down

down where? i'm still up here retarded ignorant kid @ISAKEM

**Comment by Jckf** [comment]
> down where? i'm still up here retarded ignorant kid @ISAKEM

You have been reported for harassment, and I am now locking the conversation since you insist on continuing.
